vpr/src/place/analytical_placement/AnalyticalSolver.h

```diff
@@ -26,6 +28,20 @@ class QPHybridSolver : public AnalyticalSolver {
 // need to store the matrices for the system of equations in the class.
 // TODO: Store a standard VTR matrix and pass that into Eigen using Map
 public:
-    void solve(PartialPlacement &p_placement) final;
-};
+    QPHybridSolver(PartialPlacement &p_placement);
+    void solve(unsigned iteration) final;
+    PartialPlacement &p_placement;
```

Comment:
I would not store the partial_placement object here. We do not want hidden state in these classes when we can avoid it. Although it seems a bit much, I recommend passing the partial_placement each iteration of the solver. This makes it clear that we are using the partial_placement object and modifying it.